### PR TITLE
release candidate 7

### PR DESCRIPTION
**Pull request by velzevur**

### CODE REVIEW

File: config/vm.args [hunk 1]
```diff
@@ -1,6 +1,5 @@
 -sname epoch@localhost
 
--setcookie epoch_cookie
```
Comment (dincho) [inline config/vm.args]: I hope it's planned

File: docs/release-notes/RELEASE-NOTES-1.0.0-rc7.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# About this release
+
+[This release][this-release] is the seventh release candidate.
+It:
+* New error handling for smart contracts. This affects consensus.
+* Incease gossiping of created blocks.
```
Comment (lucafavatella) [inline docs/release-notes/RELEASE-NOTES-1.0.0-rc7.md]: Typo. Increase

File: docs/release-notes/RELEASE-NOTES-1.0.0-rc7.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# About this release
+
+[This release][this-release] is the seventh release candidate.
+It:
+* New error handling for smart contracts. This affects consensus.
```
Comment (jur0) [inline docs/release-notes/RELEASE-NOTES-1.0.0-rc7.md]: "Introduces new error handling..."?

File: config/vm.args (hunk 1 above)
Comment (velzevur) [reply config/vm.args]: It is

Comment (lucafavatella) [reply config/vm.args]: Alternative:
From [OTP 20.1 doc](http://erlang.org/documentation/doc-9.1/lib/kernel-5.4/doc/html/kernel_app.html): `{inet_dist_use_interface, ip_address()}`.
Used [here](https://github.com/erlang/otp/blob/OTP-20.1/lib/kernel/src/inet_tcp_dist.erl#L117).

Comment (lucafavatella) [reply config/vm.args]: > Alternative:
> From [OTP 20.1 doc](http://erlang.org/documentation/doc-9.1/lib/kernel-5.4/doc/html/kernel_app.html): `{inet_dist_use_interface, ip_address()}`.
> Used [here](https://github.com/erlang/otp/blob/OTP-20.1/lib/kernel/src/inet_tcp_dist.erl#L117).

`net_kernel` checks `proto_dist` init argument and if none it uses `inet_tcp_dist` ([doc](https://github.com/erlang/otp/blob/OTP-20.1/erts/doc/src/erl.xml#L439-L444), [code](https://github.com/erlang/otp/blob/OTP-20.1/lib/kernel/src/net_kernel.erl#L1380-L1384)).

`net_kernel` listens calling [`inet_tcp_dist:listen(Name)`](https://github.com/erlang/otp/blob/OTP-20.1/lib/kernel/src/net_kernel.erl#L1399-L1400) that calls internal `inet_tcp_dist:do_listen` [with `Driver` `inet_tcp` (hardcoded)](https://github.com/erlang/otp/blob/OTP-20.1/lib/kernel/src/inet_tcp_dist.erl#L71-L75). This, if `application:get_env(kernel, inet_dist_use_interface)` returns `{ok, Ip}`, adds `{ip, Ip}` to options passed to `Driver:listen(First, Options)` i.e. [`gen_tcp:listen/2`](https://github.com/erlang/otp/blob/OTP-20.1/lib/kernel/src/inet_tcp.erl#L140), that in turn [parses option `{ip, _}`](https://github.com/erlang/otp/blob/OTP-20.1/lib/kernel/src/inet.erl#L802).

So option `inet_dist_use_interface` seems hooked properly.

Comment (lucafavatella) [reply config/vm.args]: Coded as https://github.com/aeternity/epoch/pull/1859 - testing

**Pull request merged**